pwiz_tools/Skyline/TestRunnerLib/RunTests.cs

```diff
@@ -370,6 +379,23 @@ public bool Run(TestInfo test, int pass, int testNumber, string dmpDir)
             else
                 ErrorCounts[failureInfo] = 1;
 
+            if (TeamCityTestDecoration)
+            {
+                // ReSharper disable LocalizableElement
+                var tcMessage = new System.Text.StringBuilder(message);
+                tcMessage.AppendLine();
+                tcMessage.AppendLine(stackTrace);
+                tcMessage.Replace("'", "|'");
+                tcMessage.Replace("\n", "|n");
+                tcMessage.Replace("\r", "|r");
+                tcMessage.Replace("|", "||");
```

Comment:
Usually, replacing "|" with "||" would be the absolutely first replacement. Otherwise, you end up replacing the "|n" with "||n", which is probably not what you want.